### PR TITLE
DOC: update the pandas.DataFrame.isna and pandas.Series.isna docstring

### PR DESCRIPTION
**Pull request by datadonK23**
Checklist for the pandas documentation sprint (ignore this if you are doing
an unrelated PR):

- [x] PR title is "DOC: update the <your-function-or-method> docstring"
- [x] The validation script passes: `scripts/validate_docstrings.py <your-function-or-method>`
- [x] The PEP8 style check passes: `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] The html version looks good: `python doc/make.py --single <your-function-or-method>`
- [x] It has been proofread on language by another sprint participant

Two Validations for pandas.DataFrame.isna and pandas.Series.isna (shared docs).

```
################################################################################
###################### Docstring (pandas.DataFrame.isna)  ######################
################################################################################

Detect missing values.

Return a boolean same-sized object indicating if the values are NA.
NA values, such as None or :attr:`numpy.NaN`, get mapped to True
values.
Everything else get mapped to False values. Characters such as empty
strings `''` or :attr:`numpy.inf` are not considered NA values
(unless you set :attr:`pandas.options.mode.use_inf_as_na` `= True`).

Returns
-------
bool of type DataFrame
    Mask of True/False values for each element in DataFrame that
    indicates whether an element is an NA value

See Also
--------
DataFrame.isnull : alias of isna
DataFrame.notna : boolean inverse of isna
DataFrame.dropna : omit axes labels with missing values
isna : top-level isna

Examples
--------
Show which entries in a DataFrame are NA.

>>> df = pd.DataFrame({'age': [5, 6, np.NaN],
...                    'born': [pd.NaT, pd.Timestamp('1939-05-27'),
...                             pd.Timestamp('1940-04-25')],
...                    'name': ['Alfred', 'Batman', ''],
...                    'toy': [None, 'Batmobile', 'Joker']})
>>> df
   age       born    name        toy
0  5.0        NaT  Alfred       None
1  6.0 1939-05-27  Batman  Batmobile
2  NaN 1940-04-25              Joker

>>> df.isna()
     age   born   name    toy
0  False   True  False   True
1  False  False  False  False
2   True  False  False  False

Show which entries in a Series are NA.

>>> ser = pd.Series([5, 6, np.NaN])
>>> ser
0    5.0
1    6.0
2    NaN
dtype: float64

>>> ser.isna()
0    False
1    False
2     True
dtype: bool

################################################################################
################################## Validation ##################################
################################################################################

Docstring for "pandas.DataFrame.isna" correct. :)
```

```
################################################################################
######################## Docstring (pandas.Series.isna) ########################
################################################################################

Detect missing values.

Return a boolean same-sized object indicating if the values are NA.
NA values, such as None or :attr:`numpy.NaN`, get mapped to True
values.
Everything else get mapped to False values. Characters such as empty
strings `''` or :attr:`numpy.inf` are not considered NA values
(unless you set :attr:`pandas.options.mode.use_inf_as_na` `= True`).

Returns
-------
bool of type Series
    Mask of True/False values for each element in Series that
    indicates whether an element is an NA value

See Also
--------
Series.isnull : alias of isna
Series.notna : boolean inverse of isna
Series.dropna : omit axes labels with missing values
isna : top-level isna

Examples
--------
Show which entries in a DataFrame are NA.

>>> df = pd.DataFrame({'age': [5, 6, np.NaN],
...                    'born': [pd.NaT, pd.Timestamp('1939-05-27'),
...                             pd.Timestamp('1940-04-25')],
...                    'name': ['Alfred', 'Batman', ''],
...                    'toy': [None, 'Batmobile', 'Joker']})
>>> df
   age       born    name        toy
0  5.0        NaT  Alfred       None
1  6.0 1939-05-27  Batman  Batmobile
2  NaN 1940-04-25              Joker

>>> df.isna()
     age   born   name    toy
0  False   True  False   True
1  False  False  False  False
2   True  False  False  False

Show which entries in a Series are NA.

>>> ser = pd.Series([5, 6, np.NaN])
>>> ser
0    5.0
1    6.0
2    NaN
dtype: float64

>>> ser.isna()
0    False
1    False
2     True
dtype: bool

################################################################################
################################## Validation ##################################
################################################################################

Docstring for "pandas.Series.isna" correct. :)

```

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -5458,13 +5458,63 @@ def asof(self, where, subset=None):
...
+        values.
+        Everything else get mapped to False values. Characters such as empty
+        strings `''` or :attr:`numpy.inf` are not considered NA values
+        (unless you set :attr:`pandas.options.mode.use_inf_as_na` `= True`).
+
+        Returns
+        -------
+        bool of type %(klass)s
+            Mask of True/False values for each element in %(klass)s that
+            indicates whether an element is an NA value
```
Comment (villasv) [inline pandas/core/generic.py]: Missing dot

File: pandas/core/indexes/base.py [hunk 2]
```diff
@@ -2022,18 +2022,48 @@ def hasnans(self):
...
+        strings `''` or :attr:`numpy.inf` are not considered NA values
+        (unless you set :attr:`pandas.options.mode.use_inf_as_na` `= True`).
 
         .. versionadded:: 0.20.0
 
         Returns
         -------
-        a boolean array of whether my values are NA
+        numpy.ndarray
+            A boolean array of whether my values are NA
```
Comment (villasv) [inline pandas/core/indexes/base.py]: Missing dot

**Review by villasv — CHANGES_REQUESTED** [review]

**Comment by donK23** [comment]
```
################################################################################
###################### Docstring (pandas.DataFrame.isna)  ######################
################################################################################

Detect missing values.

Return a boolean same-sized object indicating if the values are NA.
NA values, such as None or :attr:`numpy.NaN`, get mapped to True
values.
Everything else get mapped to False values. Characters such as empty
strings `''` or :attr:`numpy.inf` are not considered NA values
(unless you set :attr:`pandas.options.mode.use_inf_as_na` `= True`).

Returns
-------
bool of type DataFrame
    Mask of True/False values for each element in DataFrame that
    indicates whether an element is an NA value.

See Also
--------
DataFrame.isnull : alias of isna
DataFrame.notna : boolean inverse of isna
DataFrame.dropna : omit axes labels with missing values
isna : top-level isna

Examples
--------
Show which entries in a DataFrame are NA.

>>> df = pd.DataFrame({'age': [5, 6, np.NaN],
...                    'born': [pd.NaT, pd.Timestamp('1939-05-27'),
...                             pd.Timestamp('1940-04-25')],
...                    'name': ['Alfred', 'Batman', ''],
...                    'toy': [None, 'Batmobile', 'Joker']})
>>> df
   age       born    name        toy
0  5.0        NaT  Alfred       None
1  6.0 1939-05-27  Batman  Batmobile
2  NaN 1940-04-25              Joker

>>> df.isna()
     age   born   name    toy
0  False   True  False   True
1  False  False  False  False
2   True  False  False  False

Show which entries in a Series are NA.

>>> ser = pd.Series([5, 6, np.NaN])
>>> ser
0    5.0
1    6.0
2    NaN
dtype: float64

>>> ser.isna()
0    False
1    False
2     True
dtype: bool

################################################################################
################################## Validation ##################################
################################################################################

Docstring for "pandas.DataFrame.isna" correct. :)
```

File: pandas/core/indexes/base.py [hunk 3]
```diff
@@ -2022,48 +2022,18 @@ def hasnans(self):
...
-        (unless you set :attr:`pandas.options.mode.use_inf_as_na` `= True`).
+        Detect missing values
 
         .. versionadded:: 0.20.0
 
         Returns
         -------
-        numpy.ndarray
-            A boolean array of whether my values are NA
+        a boolean array of whether my values are NA
```
Comment (villasv) [inline pandas/core/indexes/base.py]: The right format should probably be
```
numpy.ndarray
    Boolean array of whether my values are NA.
```

Though I'd advise against "my values" and refer to the data structure.

> The documentation of the return is also similar to the parameters. But in this case, no name will be provided, unless the method returns or yields more than one value (a tuple of values).

> The parameters are defined by their name, followed by a space, a colon, another space, and the type (or types). Note that the space between the name and the colon is important. Types are not defined for *args and **kwargs, but must be defined for all other parameters. After the parameter definition, it is required to have a line with the parameter description, which is indented, and can have multiple lines. The description must start with a capital letter, and finish with a dot.

**Review by villasv — CHANGES_REQUESTED** [review]

File: pandas/core/indexes/base.py (hunk 3 above)
Comment (donK23) [reply pandas/core/indexes/base.py]: Please ignore, was from incorrect commit

File: pandas/core/generic.py [hunk 4]
```diff
@@ -5458,13 +5458,63 @@ def asof(self, where, subset=None):
...
 
     _shared_docs['isna'] = """
+        Detect missing values.
+
         Return a boolean same-sized object indicating if the values are NA.
+        NA values, such as None or :attr:`numpy.NaN`, get mapped to True
+        values.
+        Everything else get mapped to False values. Characters such as empty
+        strings `''` or :attr:`numpy.inf` are not considered NA values
+        (unless you set :attr:`pandas.options.mode.use_inf_as_na` `= True`).
```
Comment (jorisvandenbossche) [inline pandas/core/generic.py]: Similar comment here as on the notna PR

File: pandas/core/generic.py [hunk 5]
```diff
@@ -5458,13 +5458,63 @@ def asof(self, where, subset=None):
...
+
+        Returns
+        -------
+        bool of type %(klass)s
```
Comment (jorisvandenbossche) [inline pandas/core/generic.py]: and here as well

**Review by jorisvandenbossche — COMMENTED** [review]
Minor comment, for the rest looking very good!

**Comment by donK23** [comment]
Made the changes. Now it should be consistent with the .notna docstrings (PR #20160).

**Comment by TomAugspurger** [comment]
Thanks @donK23 !

**Pull request merged**